examples/step-89/step-89.cc

```diff
@@ -0,0 +1,1962 @@
+        Vector<Number> rho(
+          matrix_free.get_dof_handler().get_triangulation().n_active_cells());
+
+        for (const auto &cell : matrix_free.get_dof_handler()
+                                  .get_triangulation()
+                                  .active_cell_iterators())
+          {
+            c[cell->active_cell_index()] =
+              materials.at(cell->material_id()).first;
+            rho[cell->active_cell_index()] =
+              materials.at(cell->material_id()).second;
+          }
+
+        c_r->gather_evaluate(c, EvaluationFlags::values);
+        rho_r->gather_evaluate(rho, EvaluationFlags::values);
+      }
+
+    // Quadrature points are arbitrarily distributed on each non-matching
+    // face. Therefore, we have to make use of FEPointEvaluation.
+    // FEPointEvaluation needs NonMatching::MappingInfo to work at the correct
+    // quadrature points that are in sync with used FERemoteEvaluation object.
+    // In the case of mortaring, we have to use the weights provided by the
+    // quadrature rules that are used to setup NonMatching::MappingInfo.
+    // Therefore we set the flag @c use_global_weights.
+    typename NonMatching::MappingInfo<dim, dim, Number>::AdditionalData
+      additional_data;
+    additional_data.use_global_weights = true;
+
+    // Setup NonMatching::MappingInfo with needed update flags and
+    // @c additional_data.
+    auto nm_mapping_info =
+      std::make_shared<NonMatching::MappingInfo<dim, dim, Number>>(
+        mapping,
+        update_values | update_JxW_values | update_normal_vectors |
+          update_quadrature_points,
+        additional_data);
+
+    // Reinit faces with the same vector of quadratures that is used to setup
+    // the remote communicator.
+    nm_mapping_info->reinit_faces(
+      matrix_free.get_dof_handler().get_triangulation().active_cell_iterators(),
+      global_quadrature_vector);
+
+    // Setup inverse mass operator.
+    const auto inverse_mass_operator =
+      std::make_shared<InverseMassOperator<dim, Number>>(matrix_free);
+
+    // Setup the acoustic operator. Using `remote_value_type=Number`
+    // makes the operator use Nitsche-type mortaring.
+    const auto acoustic_operator =
+      std::make_shared<AcousticOperator<dim, Number, remote_value_type>>(
+        matrix_free,
+        material_data,
+        non_matching_faces,
+        pressure_r,
+        velocity_r,
+        c_r,
+        rho_r,
+        nm_mapping_info);
+
+    // Compute the the maximum speed of sound, needed for the computation of
+    // the time-step size.
+    double speed_of_sound_max = 0.0;
+    for (const auto &mat : materials)
+      speed_of_sound_max = std::max(speed_of_sound_max, mat.second.first);
+
+
+    // Setup time integrator.
+    RungeKutta2<dim, Number, remote_value_type> time_integrator(
+      inverse_mass_operator, acoustic_operator);
+
+    // Run time loop with Courant number $0.1$.
```

Comment:
Are you not specifying a Courant of 0.2 in the function call?
I would explain the rationale behind your choice of Courant number